doc/manual/types-vs-contracts.md

```diff
@@ -0,0 +1,148 @@
+# Type vs contracts: when to?
+
+You are writing Nickel code and wonder how it should be annotated. Leave it
+alone? Add type annotations? Use contracts? Here is a quick guide when you don't
+know what to do!
+
+What is the nature of the expression you are considering?
+
+## A function
+
+Most of the time, you should use **type annotations** for functions. For
+reusable code, typing is often more adapted than contracts. You can
+exceptionally use contracts when types are not expressive enough to encode the
+property you want (such as in `#ValidUrl -> #Port -> #ValidUrl`) or if the type
+system is not powerful enough to see that your code is correct.
+
+Type annotations do have a runtime cost. If you hit contracts-related
+performances issues, you can always disable some contract checks using specific
+flags (TO BE PRECISED ONCE WE HAVE THOSE FLAGS). With annotations, code is still
```

Comment:
I think that the performance discussion is out of place here.

Reply:
Removed